### ISSUE TITLE
Usage Scenario of Carbondata

### ISSUE THREAD

**Issue by summer179279** [HuaweiBigData/carbondata#836]
As being described in the wiki that Carbondata supports both column-oriented data storage and column group, which allows multiple columns to form a column group that would be stored as row format.
Does it mean that columns that involve in query conditions are stored as column format, while columns do not involve in query conditions are stored as row format.
If so, what about queries like "select *", will Carbondata stores duplicate data for those involved in query conditions?
In other words, can you explain more about the storage scheme of Carbondata?

**Comment by kumarvishal09** [comment]
Carbondata is not storing any duplicate data. Purpose of Column group is to improve the query performance. This feature is useful when  more number of columns are selected in query so reading the column data from file will faster, basically to overcome IO bottleneck. Another purpose of column group to enable user to create hybrid[columnar and row oriented] store by selecting column which will be in where condition as column oriented format and columns which will be part of select statement in column-group format.
